virtual_rainforest/core/utility_functions.py

```diff
@@ -0,0 +1,38 @@
+"""The ``core.utility_functions`` module contains functions that are used across the
+Virtual Rainforest, but which don't have a natural home in a specific module. At the
+moment, this module only contains a single function, but it will probably expand in
+future. Adding functions here can be a good way to reduce the amount boiler plate code
+generated for tasks that are repeated across modules.
+"""  # noqa: D205, D415
+
+from typing import Any
+
+import pint
+from numpy import datetime64, timedelta64
+
+
+def extract_model_time_details(
+    config: dict[str, Any], model_name: str
+) -> tuple[datetime64, timedelta64]:
+    """Function to extract the timing details required to setup a specific model.
+
+    Args:
+        config: The configuration for the Virtual Rainforest simulation.
+        model_name: The name of the specific model of interest.
+
+    Returns:
+        A tuple containing the start date and the update interval for the model
+
+    Raises:
+        pint.errors.DimensionalityError: If the desired conversion between units isn't
+            possible, e.g. kg to s.
+        pint.errors.UndefinedUnitError: If the unit is not known to pint.
+    """
+
+    raw_interval = pint.Quantity(config[model_name]["model_time_step"]).to("seconds")
+    # Round raw time interval to nearest minute
+    update_interval = timedelta64(int(round(raw_interval.magnitude)), "s")
+
+    start_date = datetime64(config["core"]["timing"]["start_date"])
+
+    return start_date, update_interval
```

Comment:
I think it might be worth moving the logging in here too? At the moment, the `from_config` methods are building up one long monolithic `try` for config errors. That works but as and when `utils` functions grow, it will probably be easier to keep some logical sections: core `util` checking, model specific checking. That also makes logging messages a bit less generic? If we did this:

```suggestion
def extract_model_time_details(
    config: dict[str, Any], model_name: str
) -> tuple[datetime64, timedelta64]:
    """Function to extract the timing details required to setup a specific model.

    Args:
        config: The configuration for the Virtual Rainforest simulation.
        model_name: The name of the specific model of interest.

    Returns:
        A tuple containing the start date and the update interval for the model

    Raises:
        pint.errors.DimensionalityError: If the desired conversion between units isn't
            possible, e.g. kg to s.
        pint.errors.UndefinedUnitError: If the unit is not known to pint.
    """

    try:
        raw_interval = pint.Quantity(config[model_name]["model_time_step"]).to("seconds")
        # Round raw time interval to nearest minute
        update_interval = timedelta64(int(round(raw_interval.magnitude)), "s")

        start_date = datetime64(config["core"]["timing"]["start_date"])

    except (pint.errors.DimensionalityError, pint.errors.UndefinedUnitError) as excep:
        LOGGER.error("Model timing error:" + excep)
        raise excep
```

Then `from_config` could all start:

```
valid_input = True
try:
    start_time, update_interval = extract_model_time_details(
        config, cls.model_name
    )
except Exception:
     valid_input = False
```

And then move on to model specific stuff? There's no way to avoid that `try` block, because of the `valid_input` sentinel. Well, unless we made `extract_model_time_details` define it:

```
valid_input,  start_time, update_interval = extract_model_time_details(
        config, cls.model_name
    )
```

Then `extract_model_time_details` would return `True, datetime64, timedelta64` or `False, None, None`. But although shorter, that seems confusing.

Reply:
Yeah I was wondering if logging would be better inside the function, agree that what you've proposed makes sense as it minimises code complexity in the individual model classes, which was kind of the point of this pull request